### DISCUSSION TITLE
@@ modifiers

### DISCUSSION THREAD

**Discussion by trufae** [radareorg/radare2#19932] [Q&A]
The current help message for `@@?` is the following:

```
[0x00000000]> @@?
| @@                   # foreach iterator command:
| x @@ sym.*          run 'x' over all flags matching 'sym.' in current flagspace
| x @@.file           run 'x' over the offsets specified in the file (one offset per line)
| x @@/x 9090         temporary set cmd.hit to run a command on each search result
| x @@=`pdf~call[0]`  run 'x' at every call offset of the current function
| x @@=off1 off2 ..   manual list of offsets
| x @@b               run 'x' on all basic blocks of current function (see afb)
| x @@c:cmd           the same as @@=`` without the backticks
| x @@dbt[abs]        run 'x' command on every backtrace address, bp or sp
| x @@f               run 'x' on all functions (see aflq)
| x @@f:write         run 'x' on all functions matching write in the name
| x @@i               run 'x' on all instructions of the current function (see pdr)
| x @@iS              run 'x' on all sections adjusting blocksize
| x @@k sdbquery      run 'x' on all offsets returned by that sdbquery
| x @@s:from to step  run 'x' on all offsets from, to incrementing by step
| x @@t               run 'x' on all threads (see dp)
```

I see some inconsistencies:

* Some modifiers have `:` while others don't.
* Some take or run a command which is redundant (@@iS vs @@c:)
* The `=` modifier is confusing as its not alphabetic char like the rest maybe should be `@@a:addr1 addr2` (or pick another letter)
* The @@k should be @@k:
* iterate over strings using r_str_glob() or using strstr ? right now we have a *hack* in 58b367610d712be9174eb70d97ea8cf0afa84681
* `f` can be confusing between flags and functions
* dbt[abs] wat

**Comment by alex-bender** [comment]
Can we change `x @@f:write` to `x @@f~write` since ~ is more like grep?

**Reply by trufae** [reply]
the problem with that is that ~affects the output of the command, so that breaks the syntax rules . f: works better imho and breaks nothing

**Reply by alex-bender** [reply]
it breaks nothing but adds inconsistency in how to grep in different contexts

**Comment by alex-bender** [comment]
Maybe that's possible to combine `s` for interchanging it from `=`, but `from to step` probably will not allow to

**Reply by trufae** [reply]
yes, the from/to/step thing is always a bit confusing to me, maybe worth discussing the syntax for it :) any proposal for this? maybe we can find a better syntax

**Comment by alex-bender** [comment]
how to use that `x @@dbt[abs]`?

**Reply by trufae** [reply]
thats not a valid syntax :? what do you mean?

**Reply by alex-bender** [reply]
![image](https://user-images.githubusercontent.com/3628862/166228354-cfd16fc0-cb11-4963-a44b-9b594abb14d8.png)
I've copied that from the help message itself
it's just unclear to me how to use that

**Comment by trufae** [comment]
omg thats pretty confusing and wrong. thanks for spotting it, which command you used to show this help? im seeing it now with ?*~dbt

**Reply by alex-bender** [reply]
> The current help message for @@? is the following:

i've took that from the first comment of this tread. so `@@?`
